edb/server/compiler/compiler.py

```diff
@@ -1898,18 +1914,26 @@ def compile(
 
         if sess_config is None:
             sess_config = EMPTY_MAP
+
+        if database_config is None:
+            database_config = EMPTY_MAP
+
+        if system_config is None:
+            system_config = EMPTY_MAP
+
         if sess_modaliases is None:
             sess_modaliases = DEFAULT_MODULE_ALIASES_MAP
 
         assert isinstance(sess_modaliases, immutables.Map)
         assert isinstance(sess_config, immutables.Map)
 
         state = dbstate.CompilerConnectionState(
-            user_schema,
-            global_schema,
-            sess_modaliases,
-            sess_config,
-            reflection_cache,
+            user_schema=user_schema,
+            global_schema=global_schema,
+            modaliases=sess_modaliases,
+            session_config=sess_config,
```

Comment:
I see you've also added it to CompilerConnectionState, not sure it's needed in CompileContext